lib/web_ui/test/engine/composition_test.dart

```diff
@@ -106,14 +106,81 @@ Future<void> testMain() async {
     });
 
     group('determine composition state', () {
-      test('should return new composition state - compositing middle of text', () {
-        const int baseOffset = 100;
+      test('should return editing state if extentOffset is null', () {
+        final EditingState editingState = EditingState(text: 'Test');
+
+        final _MockWithCompositionAwareMixin mockWithCompositionAwareMixin =
+            _MockWithCompositionAwareMixin();
+        mockWithCompositionAwareMixin.composingText = 'Test';
+
+        expect(
+            mockWithCompositionAwareMixin
+                .determineCompositionState(editingState),
+            editingState);
```

Comment:
nit: trailing comma after `editingState` and put the parentheses on the next line.